src/platform/nftPlatforms.ts

```diff
@@ -43,18 +51,27 @@ export const NFT_PLATFORM_CONFIG: { [key: string]: NFTPlatform } = {
         const data = (await response.json()) as
           | { chainId: number; contractAddress: `0x${string}`; tokenId: string }
           | undefined;
+
         if (data && data.chainId && data.contractAddress && data.tokenId) {
-          return {
-            platform: NFT_PLATFORM_CONFIG["Pods"],
-            chain: PODS_CHAIN_MAPPING[data.chainId],
-            contractAddress: data.contractAddress,
-            nftId: data.tokenId, // all are 1155s, this is the episode tokenId
-            service: new PodsService(PODS_CHAIN_MAPPING[data.chainId]),
-          } satisfies NFTExtraction;
+          if (data.chainId in PODS_CHAIN_ID_MAPPING) {
+            const chainId = data.chainId as keyof typeof PODS_CHAIN_ID_MAPPING;
+            return {
+              platform: NFT_PLATFORM_CONFIG["Pods"],
+              chain: PODS_CHAIN_ID_MAPPING[chainId],
+              contractAddress: data.contractAddress,
+              nftId: data.tokenId, // all are 1155s, this is the episode tokenId
+              service: new PodsService(PODS_CHAIN_ID_MAPPING[chainId]),
+            } satisfies NFTExtraction<PodsSupportedChain>;
+          }
         }
+        throw new Error("Failed to parse Pods URL");
       }
     },
-    platformService: PodsService,
+    /*
+     * this type coercion is necessary because the constructor signature is not compatible with
+     * the IPlatformService interface, and we're referencing it in the initializer
+     */
+    platformService: PodsService as unknown as PlatformServiceConstructor,
```

Comment:
We may get pushback on this, I don't have a problem with it personally!

Reply:
I spent 45 min trying to get more robust type inference where we limit the chains supported by our constructor, but it got *really* messy with generics and didn't fix this unless we stopped having a circular type reference. This seemed like the best option